File: backend/AniMATE_backend/AniMATE_Api/Models/Pet.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AniMATE_Api.Models;

public enum GenderType
{
    Male,
    Female
}

public class Pet
{   
    [Key]
    [StringLength(1024)]
    public string Id { get; set; } = string.Empty;
    
    [Required]
    [StringLength(256, MinimumLength = 3)]
    public string Name { get; set; } = string.Empty;
    
    [Required]
    [StringLength(256, MinimumLength = 3)]
    public string AnimalType { get; set; } = string.Empty;// TODO: Change to updatable enum
    
    [Required]
    [StringLength(256, MinimumLength = 3)]
    public string Breed { get; set; } = string.Empty;
    
    [Required]
    public int Age { get; set; }
    
    [Required]
    public GenderType Gender { get; set; }
    
    [StringLength(1024)]
    public string Description { get; set; } = string.Empty;
    

    [StringLength(1024)]
    public string Image { get; set; } = string.Empty;

    public User Owner { get; set; } = null!;
}

File: frontend/lib/models/pet.dart
import 'package:hive/hive.dart';

part 'pet.g.dart';

@HiveType(typeId: 1)
class Pet extends HiveObject{
  @HiveField(0)
  final String id;
  @HiveField(1)
  final String name;
  @HiveField(2)
  final String type;
  @HiveField(3)
  final String breed;
  @HiveField(4)
  final int age;
  @HiveField(5)
  final String gender;
  @HiveField(6)
  final String description;

  Pet({
    required this.id,
    required this.name,
    required this.type,
    required this.breed,
    required this.age,
    required this.gender,
    required this.description
  });

  factory Pet.fromJson(Map<String, dynamic> json) => Pet(
    id: json['id'],
    name: json['name'],
    type: json['animalType'],
    breed: json['breed'],
    age: json['age'],
    gender: json['gender'],
    description: json['description']
  );

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'animalType': type,
    'breed': breed,
    'age': age,
    'gender': gender,
    'description': description
  };
}